odl/util/normalize.py

```diff
@@ -262,6 +261,76 @@ def normalized_index_expression(indices, shape, int_to_slice=False):
     return tuple(indices)
 
 
+def normalized_nodes_on_bdry(nodes_on_bdry, length):
+    """Return a list of 2-tuples of bool from the input parameter.
+
+    This function is intended to normalize a ``nodes_on_bdry`` parameter
+    that can be given as a single boolean (global) or as a sequence
+    (per axis). Each entry of the sequence can either be a single
+    boolean (global for the axis) or a boolean sequence of length 2.
+
+    Parameters
+    ----------
+    nodes_on_bdry : bool or sequence
+        Input parameter to be normalized according to the above scheme.
+    length : positive int
+        Desired length of the returned list.
+
+    Returns
+    -------
+    normalized : list of 2-tuples of bool
+        Normalized list with ``length`` entries, each of which is a
+        2-tuple of boolean values.
+
+    Examples
+    --------
+    Global for all axes:
+
+    >>> normalized_nodes_on_bdry(True, length=2)
+    [(True, True), (True, True)]
+
+    Global per axis:
+
+    >>> normalized_nodes_on_bdry([True, False], length=2)
+    [(True, True), (False, False)]
+
+    Mixing global and explicit per axis:
+    >>> normalized_nodes_on_bdry([[True, False], False], length=2)
+    [(True, False), (False, False)]
+    """
+    shape = np.shape(nodes_on_bdry)
```

Comment:
`(2,)`, since it would make an `object` array from it:
```python
>>> np.array([[True, False], True])
array([[True, False], True], dtype=object)
```